generator/generate.js

```diff
@@ -536,13 +554,29 @@ ${generateFragments(name, primitiveFields, nonPrimitiveFields)}
       .join("\n")
     output += primitiveFields.length > 0 ? "\n" : ""
     output += nonPrimitiveFields
-      .map(([field, fieldName]) => {
+      .map(([field, fieldName, fieldArgs]) => {
+        let args = ""
         const selector = `${fieldName}ModelSelector`
         let p = `  ${field}(builder`
         p += ifTS(
-          `?: string | ${selector} | ((selector: ${selector}) => ${selector})`
+          `: string | ${selector} | ((selector: ${selector}) => ${selector}) | undefined`
         )
-        p += `) { return this.__child(\`${field}\`, ${selector}, builder) }`
+        if (fieldArgs && fieldArgs.length) {
+          const required = fieldArgs.find(
+            ({ type }) => type.kind === "NON_NULL"
+          )
+          p += ", args"
+          p += ifTS(
+            `${required ? "" : "?"}: { ${fieldArgs
+              .map((arg) => `${printTsType(arg)}`)
+              .join(", ")} }`
+          )
+          args = `(${fieldArgs
+            .map(({ name }) => `${name}: \${JSON.stringify(args['${name}'])}`)
+            .join(", ")})`
+          console.log(fieldArgs)
```

Comment:
Debug code

Reply:
PR #361 fixes it, already merged to main.

Reply:
great!